File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/JobApplicationResumeUserResponseDto.java
package com.ungs.docsys.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JobApplicationResumeUserResponseDto {
    private JobApplicationResponseDto jobApplication;
    private ResumeUserResponseDto resumeUser;
    private Long requirementGlobalCount;
    private Long requirementMandatoryCount;
    private Long requirementGlobalApplied;
    private Long requirementMandatoryApplied;
    private Long requirementPreferredApplied;
}

File: ungs-docsys-bff/src/modules/job-applications/job-application-resume-users/dtos/job-application-resume-user-response.dto.ts
import { ResumeUserResponseDto } from "src/modules/resume-user/dtos/resume-user-response.dto";
import { JobApplicationResponseDto } from "../../dtos/job-application-response.dto";

export class JobApplicationResumeUserResponseDto {
    jobApplication: JobApplicationResponseDto;
    resumeUser: ResumeUserResponseDto;
    requirementGlobalCount: number;
    requirementMandatoryCount: number;
    requirementGlobalApplied: number;
    requirementMandatoryApplied: number;
    requirementPreferredApplied: number;
}